bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -123,7 +123,9 @@ def run_task_with_description(task_name, targets, description, args = {})
     # executor.noop is set when run task is called from the CLI
     # options[:noop] is set when it's called from a plan
     if executor.noop || options[:noop]
-      if task.supports_noop
+      # If using the pcp transport, we don't have the task metadata. Set the noop
+      # metaparameter in this case.
+      if task.supports_noop || pcp_only
```

Comment:
I'm starting to think we should just do this in the block where we create the fake task (and alway support noop there by specifying it in the fake task metadata). That better encapsulates that bit of logic and does not bleed in to this part of the code as much.

Reply:
That makes way more sense. :) Done.